src/ui/public/vis/components/tooltip/tooltip.js

```diff
@@ -4,36 +4,62 @@ import { Binder } from 'ui/binder';
 import { positionTooltip } from './position_tooltip';
 import $ from 'jquery';
 
-let allContents = [];
-
 /**
  * Add tooltip and listeners to visualization elements
  *
+ * All Tooltip instances populate a shared div element
+ * Each Tooltip instance with unique id manages a content container within the shared div element
+ *
  * @class Tooltip
  * @constructor
  * @param el {HTMLElement} Reference to DOM element
  * @param formatter {Function} Tooltip formatter
  * @param events {Constructor} Allows tooltip to return event response data
+ * @param options {Object} Tooltip options
  */
-export function Tooltip(id, el, formatter, events) {
+export function Tooltip(id, el, formatter, events, options = {}) {
   if (!(this instanceof Tooltip)) {
-    return new Tooltip(id, el, formatter, events);
+    return new Tooltip(id, el, formatter, events, options);
   }
 
   this.id = id; // unique id for this tooltip type
   this.el = el;
-  this.order = 100; // higher ordered contents are rendered below the others
   this.formatter = formatter;
   this.events = events;
   this.containerClass = 'vis-wrapper';
   this.tooltipClass = 'vis-tooltip';
   this.tooltipSizerClass = 'vis-tooltip-sizing-clone';
-  this.showCondition = _.constant(true);
+  this.order = _.get(options, 'order', 100); // higher ordered contents are rendered below the others
+  this.showCondition = _.get(options, 'showCondition', _.constant(true));
+  this.updateContentOnMove = _.get(options, 'updateContentOnMove', false);
+  this.contentContainer = this.getContentContainer();
+  this.enterCount = 0;
 
   this.binder = new Binder();
 }
 
 /**
+ * Get content container for Tooltip.
+ * Tooltips with the same id share a content container.
+ */
+Tooltip.prototype.getContentContainer = function () {
```

Comment:
The old implementation just concatenated a bunch of HTML strings together. Each mousemove event destroyed the existing HTML and generated new HTML. 

This is no longer possible since the HTML can now be generated by asynchronous calls. Putting each tooltip in its own container helped make the implementation more straightforward.